nce/nce_pairwise_mp/trainers/qa_trainer.py

```diff
@@ -171,19 +171,18 @@ def train_epoch(self, epoch):
                     loss.backward()
                     self.optimizer.step()
 
-                    del output
-                    del loss
                     del new_train_neg
```

Comment:
Deleting `new_train_neg` and `new_train_pos` will have an impact?

Reply:
They contain my self-defined objects with GPU memory. Using del might be safer but I am not clear how it works in the low level environment. 
https://discuss.pytorch.org/t/tensor-to-variable-and-memory-freeing-best-practices/6000/2